File: Client/src/app/shared/models/order.ts
export interface Order {
  id: number;
  orderDate: string;
  buyerEmail: string;
  shippingAddress: ShippingAddress;
  deliveryMethod: string;
  shippingPrice: number;
  paymentSummary: PaymentSummary;
  orderItems: OrderItem[];
  subTotal: number;
  discount?: number;
  orderStatus: string;
  paymentIntentId: string;
  total: number;
}

export interface ShippingAddress {
  name: string;
  line1: string;
  line2?: string;
  country: string;
  city: string;
  state: string;
  postalCode: string;
}

export interface PaymentSummary {
  last4: number;
  brand: string;
  expMonth: number;
  expYear: number;
}

export interface OrderItem {
  productId: number;
  productName: string;
  pictureUrl: string;
  price: number;
  quantity: number;
}

export interface orderToCreate {
  cartId: string;
  deliveryMethodId: number;
  shippingAddress: ShippingAddress;
  paymentSummary: PaymentSummary;
  discount?: number;
}

File: Server/Core/Models/OrderAggregate/Order.cs
namespace Core.Models.OrderAggregate;

public class Order:BaseModel
{
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    public required string BuyerEmail { get; set; }
    public ShippingAddress ShippingAddress { get; set; } = null!;
    public DeliveryMethod DeliveryMethod { get; set; } = null!;
    public PaymentSummary PaymentSummary { get; set; }=null!;
    public List<OrderItem> OrderItems { get; set; } = [];
    public decimal SubTotal { get; set; }
    public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
    public required string PaymentIntentId { get; set; }
    public decimal Discount { get; set; }

    public decimal GetTotal()
    {
        
        return SubTotal - Discount + DeliveryMethod.Price;
    }


}
